packages/components/time-picker/src/common/picker.vue

```diff
@@ -367,7 +367,11 @@ const handleFocusInput = (e?: FocusEvent) => {
   ) {
     return
   }
-  pickerVisible.value = true
+  if (e?.relatedTarget == null) {
+    pickerVisible.value = true
+  } else {
+    pickerVisible.value = false
+  }
```

Comment:
Maybe we can simplify the logic via `isNil` from lodash:

```ts
import { isNil } from 'lodash-unified'
```

```ts
pickerVisible.value = isNil(e?.relatedTarget)
```

Reply:
Good idea, thanks a lot!